File: app/src/main/java/com/yyide/chatim/activity/leave/AskForLeaveActivity.java
package com.yyide.chatim.activity.leave;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.viewpager2.widget.ViewPager2;

import android.os.Bundle;
import android.widget.TextView;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;
import com.yyide.chatim.R;
import com.yyide.chatim.SpData;
import com.yyide.chatim.base.BaseActivity;
import com.yyide.chatim.base.BaseMvpFragment;
import com.yyide.chatim.fragment.leave.AskForLeaveListFragment;
import com.yyide.chatim.fragment.leave.RequestLeaveStaffFragment;
import com.yyide.chatim.fragment.leave.RequestLeaveStudentFragment;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;
import lombok.val;

public class AskForLeaveActivity extends BaseActivity {
    @BindView(R.id.title)
    TextView title;
    @BindView(R.id.tablayout)
    TabLayout mTablayout;
    @BindView(R.id.viewpager)
    ViewPager2 mViewpager;
    List<BaseMvpFragment> fragments = new ArrayList<>();
    @Override
    public int getContentViewID() {
        return R.layout.activity_ask_for_leave;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        title.setText(R.string.ask_for_leave);
        initViewPager();
    }

    @OnClick(R.id.back_layout)
    public void click() {
        finish();
    }

    private void initViewPager() {
        if (SpData.getIdentityInfo().staffIdentity()) {
            //教职工入口
            fragments.add(RequestLeaveStaffFragment.newInstance("staff ask for leave"));
        }else {
            fragments.add(RequestLeaveStudentFragment.newInstance("student ask for leave"));
        }
        fragments.add(AskForLeaveListFragment.newInstance("ask for leave record"));
        mViewpager.setOrientation(ViewPager2.ORIENTATION_HORIZONTAL);
        mViewpager.setAdapter(new FragmentStateAdapter(this) {
            @NonNull
            @Override
            public Fragment createFragment(int position) {
                if (position == 0) {
                    return fragments.get(0);
                } else {
                    return fragments.get(1);
                }
            }

            @Override
            public int getItemCount() {
                return 2;
            }
        });
        new TabLayoutMediator(mTablayout, mViewpager, true, new TabLayoutMediator.TabConfigurationStrategy() {
            @Override
            public void onConfigureTab(@NonNull TabLayout.Tab tab, int position) {
                //这里需要根据position修改tab的样式和文字等
                if (position == 0) {
                    tab.setText(R.string.my_ask_for_leave);
                } else {
                    tab.setText(R.string.leave_record);
                }
            }
        }).attach();
    }
}

File: app/src/main/java/com/yyide/chatim/activity/leave/AskForLeaveActivity.kt
package com.yyide.chatim.activity.leave

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentStatePagerAdapter
import butterknife.OnClick
import com.yyide.chatim.R
import com.yyide.chatim.base.BaseMvpFragment
import com.yyide.chatim.base.KTBaseActivity
import com.yyide.chatim.databinding.ActivityAskForLeaveBinding
import com.yyide.chatim.fragment.leave.AskForLeaveListFragment
import com.yyide.chatim.fragment.leave.RequestLeaveStaffFragment

/**
 * 请假
 */
class AskForLeaveActivity :
    KTBaseActivity<ActivityAskForLeaveBinding>(ActivityAskForLeaveBinding::inflate) {

    var fragments: MutableList<BaseMvpFragment<*>> = ArrayList()
    override fun getContentViewID(): Int {
        return R.layout.activity_ask_for_leave
    }

    override fun initView() {
        super.initView()
        binding.top.title.setText(R.string.ask_for_leave)
        binding.top.backLayout.setOnClickListener { finish() }
        initViewPager()
    }

    @OnClick(R.id.back_layout)
    fun click() {
        finish()
    }

    private fun initViewPager() {
        fragments.add(RequestLeaveStaffFragment.newInstance("staff ask for leave"))
        fragments.add(AskForLeaveListFragment.newInstance("ask for leave record"))
        val mTitles: MutableList<String> = ArrayList()
        mTitles.add("我要请假")
        mTitles.add("请假记录")
        binding.viewpager.offscreenPageLimit = fragments.size
        //需要动态数据设置该tab
        binding.viewpager.adapter = object :
            FragmentStatePagerAdapter(
                supportFragmentManager,
                BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
            ) {
            override fun getItem(position: Int): Fragment {
                return fragments[position]
            }

            override fun getCount(): Int {
                return mTitles.size
            }

            override fun getPageTitle(position: Int): CharSequence? {
                return mTitles[position]
            }
        }
        binding.slidingTabLayout.setViewPager(binding.viewpager)
        binding.slidingTabLayout.currentTab = 0
        binding.slidingTabLayout.notifyDataSetChanged()
    }
}